test/TransitivePathTest.cpp

```diff
@@ -251,25 +317,60 @@ TEST_P(TransitivePathTest, idToLeftBound) {
     auto T = makePathLeftBound(
         sub.clone(), {Variable{"?start"}, Variable{"?target"}},
         leftOpTable.clone(), 1, {Variable{"?x"}, Variable{"?start"}}, left,
+        right, 0, std::numeric_limits<size_t>::max(), false);
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+  {
+    auto T = makePathLeftBound(
+        sub.clone(), {Variable{"?start"}, Variable{"?target"}},
+        leftOpTable.clone(), 1, {std::nullopt, Variable{"?start"}}, left, right,
+        0, std::numeric_limits<size_t>::max(), false);
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+  {
+    auto T = makePathLeftBound(
+        sub.clone(), {Variable{"?start"}, Variable{"?target"}},
+        split(leftOpTable), 1, {Variable{"?x"}, Variable{"?start"}}, left,
         right, 0, std::numeric_limits<size_t>::max());
 
-    auto resultTable = T->computeResultOnlyForTesting();
-    ASSERT_THAT(resultTable.idTable(),
-                ::testing::UnorderedElementsAreArray(expected));
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+  {
+    auto T = makePathLeftBound(
+        sub.clone(), {Variable{"?start"}, Variable{"?target"}},
+        split(leftOpTable), 1, {std::nullopt, Variable{"?start"}}, left, right,
+        0, std::numeric_limits<size_t>::max());
+
+    auto resultTable = T->computeResultOnlyForTesting(requestLaziness());
+    assertResultMatchesIdTable(resultTable, expected);
+  }
+  {
+    auto T = makePathLeftBound(
+        sub.clone(), {Variable{"?start"}, Variable{"?target"}},
+        leftOpTable.clone(), 1, {Variable{"?x"}, Variable{"?start"}}, left,
+        right, 0, std::numeric_limits<size_t>::max(), true);
```

Comment:
Here we also have a lot of code duplication, reduce it by binding the constant parameters in a lambda etc, and add a short comment, what the tests test.